src/attrs.rs

```diff
@@ -204,6 +234,12 @@ impl<'a> Attrs<'a> {
         self
     }
 
+    /// Set [FontFeatures]
+    pub fn font_features(mut self, font_features: FontFeatures) -> Self {
+        self.font_features = font_features;
+        self
+    }
+
     /// Check if font matches
     pub fn matches(&self, face: &fontdb::FaceInfo) -> bool {
```

Comment:
Should the matches function be checking font features? This is used to determine what is grouped into a run for shaping, so it seems important.